Linting errors under 'frontend/e2e'

```
➜ pnpm lint e2e | grep -v warning  

> frontend@0.1.0 lint /home/pluto-atom-4/Documents/stoke-full-stack/react-grapql-playground/frontend
> eslint . --ext .ts,.tsx e2e

/home/pluto-atom-4/Documents/stoke-full-stack/react-grapql-playground/frontend/e2e/tests/event-bus/build-workflow.spec.ts
  84:5  error  'authenticatedPage' is defined but never used. Allowed unused args must match /^_/u  @typescript-eslint/no-unused-vars

/home/pluto-atom-4/Documents/stoke-full-stack/react-grapql-playground/frontend/e2e/tests/event-bus/event-deduplication.spec.ts
   61:13  error    Unsafe assignment of an `any` value                                                  @typescript-eslint/no-unsafe-assignment
   64:15  error    Unsafe assignment of an `any` value                                                  @typescript-eslint/no-unsafe-assignment
   65:9   error    Unsafe return of a value of type `any`                                               @typescript-eslint/no-unsafe-return
   65:18  error    Unsafe member access .__E2E_DEDUP_METRICS__ on an `any` value                        @typescript-eslint/no-unsafe-member-access
   70:14  error    'err' is defined but never used                                                      @typescript-eslint/no-unused-vars
  162:5   error    'authenticatedPage' is defined but never used. Allowed unused args must match /^_/u  @typescript-eslint/no-unused-vars
  214:5   error    'authenticatedPage' is defined but never used. Allowed unused args must match /^_/u  @typescript-eslint/no-unused-vars
  293:5   error    'authenticatedPage' is defined but never used. Allowed unused args must match /^_/u  @typescript-eslint/no-unused-vars

/home/pluto-atom-4/Documents/stoke-full-stack/react-grapql-playground/frontend/e2e/tests/event-bus/multi-client-sync.spec.ts
   17:24  error  'devices' is defined but never used. Allowed unused vars must match /^_/u             @typescript-eslint/no-unused-vars
   17:33  error  'Browser' is defined but never used. Allowed unused vars must match /^_/u             @typescript-eslint/no-unused-vars
   20:32  error  Unexpected any. Specify a different type                                              @typescript-eslint/no-explicit-any
   21:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   21:14  error  Unsafe member access .goto on an `any` value                                          @typescript-eslint/no-unsafe-member-access
   22:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   22:14  error  Unsafe member access .waitForTimeout on an `any` value                                @typescript-eslint/no-unsafe-member-access
   24:9   error  Unsafe assignment of an `any` value                                                   @typescript-eslint/no-unsafe-assignment
   24:21  error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   24:26  error  Unsafe member access .locator on an `any` value                                       @typescript-eslint/no-unsafe-member-access
   25:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   25:19  error  Unsafe member access .waitFor on an `any` value                                       @typescript-eslint/no-unsafe-member-access
   26:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   26:19  error  Unsafe member access .click on an `any` value                                         @typescript-eslint/no-unsafe-member-access
   28:9   error  Unsafe assignment of an `any` value                                                   @typescript-eslint/no-unsafe-assignment
   28:22  error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   28:27  error  Unsafe member access .locator on an `any` value                                       @typescript-eslint/no-unsafe-member-access
   29:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   29:20  error  Unsafe member access .waitFor on an `any` value                                       @typescript-eslint/no-unsafe-member-access
   36:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   36:20  error  Unsafe member access .fill on an `any` value                                          @typescript-eslint/no-unsafe-member-access
   37:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   37:20  error  Unsafe member access .blur on an `any` value                                          @typescript-eslint/no-unsafe-member-access
   38:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   38:14  error  Unsafe member access .waitForTimeout on an `any` value                                @typescript-eslint/no-unsafe-member-access
   40:9   error  Unsafe assignment of an `any` value                                                   @typescript-eslint/no-unsafe-assignment
   40:25  error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   40:30  error  Unsafe member access .locator on an `any` value                                       @typescript-eslint/no-unsafe-member-access
   41:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   41:23  error  Unsafe member access .waitFor on an `any` value                                       @typescript-eslint/no-unsafe-member-access
   42:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   42:23  error  Unsafe member access .fill on an `any` value                                          @typescript-eslint/no-unsafe-member-access
   43:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   43:23  error  Unsafe member access .blur on an `any` value                                          @typescript-eslint/no-unsafe-member-access
   44:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   44:14  error  Unsafe member access .waitForTimeout on an `any` value                                @typescript-eslint/no-unsafe-member-access
   46:9   error  Unsafe assignment of an `any` value                                                   @typescript-eslint/no-unsafe-assignment
   46:24  error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   46:29  error  Unsafe member access .locator on an `any` value                                       @typescript-eslint/no-unsafe-member-access
   47:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   47:22  error  Unsafe member access .waitFor on an `any` value                                       @typescript-eslint/no-unsafe-member-access
   49:9   error  Unsafe assignment of an `any` value                                                   @typescript-eslint/no-unsafe-assignment
   49:28  error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   49:33  error  Unsafe member access .waitForURL on an `any` value                                    @typescript-eslint/no-unsafe-member-access
   50:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   50:22  error  Unsafe member access .click on an `any` value                                         @typescript-eslint/no-unsafe-member-access
   55:51  error  Invalid type "unknown" of template literal expression                                 @typescript-eslint/restrict-template-expressions
   59:9   error  Unsafe call of an `any` typed value                                                   @typescript-eslint/no-unsafe-call
   59:14  error  Unsafe member access .waitForFunction on an `any` value                               @typescript-eslint/no-unsafe-member-access
  234:77  error  'browser' is defined but never used. Allowed unused args must match /^_/u             @typescript-eslint/no-unused-vars
  336:15  error  'buildName' is assigned a value but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/home/pluto-atom-4/Documents/stoke-full-stack/react-grapql-playground/frontend/e2e/tests/event-bus/real-time-updates.spec.ts
   38:11  error    'startTime' is assigned a value but never used. Allowed unused vars must match /^_/u     @typescript-eslint/no-unused-vars
   88:5   error    'authenticatedPage' is defined but never used. Allowed unused args must match /^_/u      @typescript-eslint/no-unused-vars
  266:13  error    'testFilePath' is assigned a value but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
  272:19  error    'Buffer' is not defined                                                                  no-undef
  289:16  error    'err' is defined but never used                                                          @typescript-eslint/no-unused-vars

/home/pluto-atom-4/Documents/stoke-full-stack/react-grapql-playground/frontend/e2e/tests/event-bus/sse-reconnection.spec.ts
   49:14  error  'err' is defined but never used                                                     @typescript-eslint/no-unused-vars
  157:70  error  Invalid type "unknown" of template literal expression                               @typescript-eslint/restrict-template-expressions
  173:11  error  'builds1' is assigned a value but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/home/pluto-atom-4/Documents/stoke-full-stack/react-grapql-playground/frontend/lib/__tests__/issue-6-integration-flows.test.ts

/home/pluto-atom-4/Documents/stoke-full-stack/react-grapql-playground/frontend/lib/__tests__/sse-stress-testing.test.ts
   20:32  error    'beforeEach' is defined but never used. Allowed unused vars must match /^_/u       @typescript-eslint/no-unused-vars
   20:44  error    'afterEach' is defined but never used. Allowed unused vars must match /^_/u        @typescript-eslint/no-unused-vars
   20:55  error    'vi' is defined but never used. Allowed unused vars must match /^_/u               @typescript-eslint/no-unused-vars
   97:35  error    'setTimeout' is not defined                                                        no-undef
  217:9   error    'global' is not defined                                                            no-undef
  218:7   error    'global' is not defined                                                            no-undef
  345:23  error    'performance' is not defined                                                       no-undef
  352:23  error    'performance' is not defined                                                       no-undef
  416:27  error    'performance' is not defined                                                       no-undef
  425:24  error    'performance' is not defined                                                       no-undef
  428:30  error    'performance' is not defined                                                       no-undef
  445:27  error    'performance' is not defined                                                       no-undef


```

--- a/frontend/e2e/helpers/api-client.ts
+++ b/frontend/e2e/helpers/api-client.ts
@@ -74,7 +74,7 @@ export class GraphQLClient {
     try {
       // Log for debugging
       const hasToken = !!this.token;
-      console.log(
+      console.warn(
         `[GraphQLClient] Executing GraphQL request with auth: ${hasToken ? 'YES' : 'NO'}`
       );
       if (!hasToken) {

--- a/frontend/e2e/helpers/seed-data.ts
+++ b/frontend/e2e/helpers/seed-data.ts
@@ -120,7 +120,7 @@ export function cleanupTestData(_apiClient: GraphQLClient, data: SeededTestData)
     // Note: Apollo GraphQL backend doesn't have delete mutations implemented
     // Test cleanup would require database-level cleanup
     // For now, just log that cleanup was requested
-    console.log('Cleanup requested for test data:', {
+    console.warn('Cleanup requested for test data:', {
       builds: data.buildIds,
       parts: data.partIds,
       testRuns: data.testRunIds,

--- a/frontend/e2e/helpers/wait-helpers.ts
+++ b/frontend/e2e/helpers/wait-helpers.ts
@@ -83,7 +83,7 @@ export async function waitForSSEEvent(
           const data = JSON.parse(messageEvent.data) as { type?: string };
           if (data.type === event) {
             // eslint-disable-next-line no-console
-            console.log(`SSE:${event}:${JSON.stringify(data)}`);
+            console.warn(`SSE:${event}:${JSON.stringify(data)}`);
             eventSource.close();
           }
         };

--- a/frontend/e2e/playwright.global-setup.ts
+++ b/frontend/e2e/playwright.global-setup.ts
@@ -2,7 +2,7 @@
  * Global setup for Playwright tests
  * Verifies all required services are running before tests begin
  */
-async function globalSetup() {
+async function globalSetup(): Promise<void> {
   console.log('🔍 Verifying service connectivity...');
 
   const serviceChecks = [

--- a/frontend/e2e/tests/event-bus/build-workflow.spec.ts
+++ b/frontend/e2e/tests/event-bus/build-workflow.spec.ts
@@ -81,7 +81,7 @@ test.describe('Event Bus: Build Workflow', () => {
   // TC-E2E-BW-002: Create Build workflow with real-time update
   // --------------------------------------------------------------------------
   test('TC-E2E-BW-002: Create Build workflow with real-time update', async ({
-    authenticatedPage,
+    _authenticatedPage,
   }) => {
     const buildName = `E2E Test Build ${Date.now()}`;
 

--- a/frontend/e2e/tests/event-bus/event-deduplication.spec.ts
+++ b/frontend/e2e/tests/event-bus/event-deduplication.spec.ts
@@ -58,13 +58,15 @@ test.describe('Event Bus: Event Deduplication', () => {
 
     // Get event listener metadata (if available) from Apollo cache
     try {
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
-      const eventMetrics = (await authenticatedPage.evaluate(() => {
-        // Try to access dedup metrics from window or Apollo cache
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const w = window as any;
-        return w.__E2E_DEDUP_METRICS__ || { rejected: 0, accepted: 0 };
-      })) as { rejected: number; accepted: number };
+      const eventMetrics = (
+        await authenticatedPage.evaluate(() => {
+          // Try to access dedup metrics from window or Apollo cache
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const w = window as any; // eslint-disable-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return
+          return w.__E2E_DEDUP_METRICS__ || { rejected: 0, accepted: 0 };
+        })
+      ) as { rejected: number; accepted: number };
 
       // Verify dedup metrics show some rejections (if implemented)
       console.warn('[TC-E2E-DEDUP-001] Dedup metrics:', eventMetrics);

--- a/frontend/e2e/tests/event-bus/event-deduplication.spec.ts
+++ b/frontend/e2e/tests/event-bus/event-deduplication.spec.ts
@@ -58,6 +58,7 @@ test.describe('Event Bus: Event Deduplication', () => {
 
     // Get event listener metadata (if available) from Apollo cache
     try {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
       const eventMetrics = (await authenticatedPage.evaluate(() => {
         // Try to access dedup metrics from window or Apollo cache
         // eslint-disable-next-line @typescript-eslint/no-explicit-any

--- a/frontend/e2e/tests/event-bus/event-deduplication.spec.ts
+++ b/frontend/e2e/tests/event-bus/event-deduplication.spec.ts
@@ -58,18 +58,18 @@ test.describe('Event Bus: Event Deduplication', () => {
 
     // Get event listener metadata (if available) from Apollo cache
     try {
-      const eventMetrics = await authenticatedPage.evaluate(() => {
+      const eventMetrics = (await authenticatedPage.evaluate(() => {
         // Try to access dedup metrics from window or Apollo cache
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
         const w = window as any;
         return w.__E2E_DEDUP_METRICS__ || { rejected: 0, accepted: 0 };
-      });
+      })) as { rejected: number; accepted: number };
 
       // Verify dedup metrics show some rejections (if implemented)
-      console.log('[TC-E2E-DEDUP-001] Dedup metrics:', eventMetrics);
-    } catch (err) {
+      console.warn('[TC-E2E-DEDUP-001] Dedup metrics:', eventMetrics);
+    } catch {
       // Metrics might not be exposed, that's ok for E2E
-      console.log('[TC-E2E-DEDUP-001] Could not retrieve dedup metrics');
+      console.warn('[TC-E2E-DEDUP-001] Could not retrieve dedup metrics');
     }
 
     // Verify: Build still appears exactly once
@@ -159,7 +159,7 @@ test.describe('Event Bus: Event Deduplication', () => {
   // TC-E2E-DEDUP-003: Dedup window expiration allows reacceptance
   // --------------------------------------------------------------------------
   test('TC-E2E-DEDUP-003: Dedup window expiration (simulated)', async ({
-    authenticatedPage,
+    _authenticatedPage,
   }) => {
     const buildName = `E2E Dedup Expiry ${Date.now()}`;
 
@@ -211,7 +211,7 @@ test.describe('Event Bus: Event Deduplication', () => {
   // TC-E2E-DEDUP-004: Idempotent operations don't cause duplicates
   // --------------------------------------------------------------------------
   test('TC-E2E-DEDUP-004: Idempotent operations prevent duplicates', async ({
-    authenticatedPage,
+    _authenticatedPage,
   }) => {
     const buildName = `E2E Idempotent ${Date.now()}`;
 
@@ -290,7 +290,7 @@ test.describe('Event Bus: Event Deduplication', () => {
   // TC-E2E-DEDUP-006: Cache consistency after dedup
   // --------------------------------------------------------------------------
   test('TC-E2E-DEDUP-006: Cache remains consistent after dedup operations', async ({
-    authenticatedPage,
+    _authenticatedPage,
   }) => {
     const buildName = `E2E Cache Consistency ${Date.now()}`;
 

--- a/frontend/e2e/tests/event-bus/multi-client-sync.spec.ts
+++ b/frontend/e2e/tests/event-bus/multi-client-sync.spec.ts
@@ -14,10 +14,10 @@
  * - No page refresh needed for synchronization
  */
 
-import { test, expect, devices, Browser } from '@playwright/test';
+import { test, expect, type Page } from '@playwright/test';
 
 // Helper to login in a page
-async function loginPage(page: any, baseURL: string): Promise<void> {
+async function loginPage(page: Page, baseURL: string): Promise<void> {
   await page.goto(`${baseURL}/`, { waitUntil: 'domcontentloaded' });
   await page.waitForTimeout(300);
 
@@ -52,7 +52,7 @@ async function loginPage(page: any, baseURL: string): Promise<void> {
   try {
     await urlChangePromise;
   } catch (err) {
-    throw new Error(`Dashboard redirect failed: ${err instanceof Error ? err.message : err}`);
+    throw new Error(`Dashboard redirect failed: ${err instanceof Error ? err.message : String(err)}`);
   }
 
   // Wait for JWT token
@@ -231,7 +231,7 @@ test.describe('Event Bus: Multi-Client Synchronization', () => {
   // --------------------------------------------------------------------------
   // TC-E2E-MCS-003: One user, two tabs synchronization
   // --------------------------------------------------------------------------
-  test('TC-E2E-MCS-003: One user, two tabs synchronization', async ({ page, browser }) => {
+  test('TC-E2E-MCS-003: One user, two tabs synchronization', async ({ page, _browser }) => {
     const baseURL = process.env.BASE_URL || 'http://localhost:3000';
 
     // Login in first page
@@ -332,8 +332,8 @@ test.describe('Event Bus: Multi-Client Synchronization', () => {
       }
 
       // Verify all 3 builds appear in page2 (they should arrive via SSE)
-      for (let i = 0; i < 3; i++) {
-        const buildName = `E2E Load Build`;
+      for (let _i = 0; _i < 3; _i++) {
+        const _buildName = `E2E Load Build`;
         const buildCards = page2.locator('[data-testid^="build-"]');
         const count = await buildCards.count();
         // Just verify we have multiple builds (exact match may vary based on test data)

--- a/frontend/e2e/tests/event-bus/real-time-updates.spec.ts
+++ b/frontend/e2e/tests/event-bus/real-time-updates.spec.ts
@@ -18,6 +18,7 @@
 
 import { test, expect } from '../../fixtures';
 import { DashboardPage } from '../../pages';
+import { Buffer } from 'node:buffer';
 
 test.describe('Event Bus: Real-Time UI Updates', () => {
   let dashboardPage: DashboardPage;
@@ -35,7 +36,7 @@ test.describe('Event Bus: Real-Time UI Updates', () => {
     authenticatedPage,
   }) => {
     const buildName = `E2E Status Update ${Date.now()}`;
-    const startTime = Date.now();
+    const _startTime = Date.now();
 
     // Create a build
     await dashboardPage.clickByTestId('create-build-button');
@@ -85,7 +86,7 @@ test.describe('Event Bus: Real-Time UI Updates', () => {
   // TC-E2E-RTUI-002: New build appears in dashboard
   // --------------------------------------------------------------------------
   test('TC-E2E-RTUI-002: New build appears in dashboard without refresh', async ({
-    authenticatedPage,
+    _authenticatedPage,
   }) => {
     await dashboardPage.goto();
 
@@ -263,7 +264,7 @@ test.describe('Event Bus: Real-Time UI Updates', () => {
       const uploadStartTime = Date.now();
 
       // Create a simple test file
-      const testFilePath = '/tmp/test-report.txt';
+      const _testFilePath = '/tmp/test-report.txt';
       try {
         // Use page to upload file
         await fileInput.setInputFiles({
@@ -286,9 +287,9 @@ test.describe('Event Bus: Real-Time UI Updates', () => {
 
         // Verify: Latency acceptable
         expect(uploadLatency).toBeLessThan(2000);
-      } catch (err) {
+      } catch {
         // File upload not implemented, that's ok for E2E test
-        console.log('File upload test skipped (feature not implemented)');
+        console.warn('File upload test skipped (feature not implemented)');
       }
     }
   });

--- a/frontend/e2e/tests/event-bus/sse-reconnection.spec.ts
+++ b/frontend/e2e/tests/event-bus/sse-reconnection.spec.ts
@@ -46,7 +46,7 @@ test.describe('Event Bus: SSE Reconnection Resilience', () => {
     try {
       await authenticatedPage.evaluate(() => fetch('http://localhost:3000/dashboard'));
       // If we get here, network is still on (test might not work on this platform)
-    } catch (err) {
+    } catch {
       // Expected: network is off
     }
 
@@ -170,7 +170,7 @@ test.describe('Event Bus: SSE Reconnection Resilience', () => {
     await dashboardPage.isDashboardReady();
 
     // Verify initial state
-    const builds1 = await dashboardPage.getBuilds();
+    const _builds1 = await dashboardPage.getBuilds();
 
     // Simulate network hiccup
     await context.setOffline(true);

--- a/frontend/e2e/tests/event-bus/sse-reconnection.spec.ts
+++ b/frontend/e2e/tests/event-bus/sse-reconnection.spec.ts
@@ -154,7 +154,7 @@ test.describe('Event Bus: SSE Reconnection Resilience', () => {
       expect(true).toBe(true);
     } catch (err) {
       // If this fails, the stress test may have broken the connection
-      throw new Error(`Dashboard not functional after stress test: ${err instanceof Error ? err.message : err}`);
+      throw new Error(`Dashboard not functional after stress test: ${err instanceof Error ? err.message : String(err)}`);
     }
   });
 

--- a/frontend/lib/__tests__/issue-6-integration-flows.test.ts
+++ b/frontend/lib/__tests__/issue-6-integration-flows.test.ts
@@ -230,7 +230,7 @@ describe('Test Suite 2: Error Scenarios with Automatic Retry', () => {
     // 3. Exponential backoff applied (100ms, 200ms, 400ms)
     // 4. Retry attempt made
 
-    const isNetworkError = (error: Error) => {
+    const isNetworkError = (error: Error): boolean => {
       return error.message.includes('Network') || error.message.includes('timeout');
     };
 
@@ -241,7 +241,7 @@ describe('Test Suite 2: Error Scenarios with Automatic Retry', () => {
   it('E2.2: Exponential backoff formula correctly calculates delays', () => {
     // Exponential backoff: delay = 100ms × 2^attempt + random jitter (±20%)
 
-    const calculateBackoff = (attempt: number, baseDelay = 100) => {
+    const calculateBackoff = (attempt: number, baseDelay = 100): number => {
       const delay = baseDelay * Math.pow(2, attempt);
       const jitter = Math.random() * 0.4 - 0.2; // ±20%
       return Math.round(delay * (1 + jitter));
@@ -264,7 +264,7 @@ describe('Test Suite 2: Error Scenarios with Automatic Retry', () => {
     // Retryable: 5xx (500, 502, 503, 504), network errors, timeouts, 429
     // Non-retryable: 4xx (except 429), 401, 403
 
-    const isRetryable = (statusCode: number) => {
+    const isRetryable = (statusCode: number): boolean => {
       if (statusCode >= 500) return true; // 5xx
       if (statusCode === 429) return true; // Rate limit
       return false;
@@ -278,7 +278,7 @@ describe('Test Suite 2: Error Scenarios with Automatic Retry', () => {
   });
 
   it('E2.4: 4xx errors (except 429) are NOT retried', () => {
-    const isRetryable = (statusCode: number) => {
+    const isRetryable = (statusCode: number): boolean => {
       if (statusCode >= 500) return true;
       if (statusCode === 429) return true;
       return false;
@@ -291,7 +291,7 @@ describe('Test Suite 2: Error Scenarios with Automatic Retry', () => {
   });
 
   it('E2.5: Timeout errors trigger retry with backoff', () => {
-    const isTimeoutError = (error: Error) => {
+    const isTimeoutError = (error: Error): boolean => {
       return error.message.includes('timeout') || error.message.includes('TIMEOUT');
     };
 
@@ -300,7 +300,7 @@ describe('Test Suite 2: Error Scenarios with Automatic Retry', () => {
   });
 
   it('E2.6: 401 (auth) errors do NOT retry, show error message', () => {
-    const shouldRetry = (statusCode: number) => {
+    const shouldRetry = (statusCode: number): boolean => {
       if (statusCode === 401) return false; // Auth error
       return true; // Other errors might retry
     };
@@ -309,7 +309,7 @@ describe('Test Suite 2: Error Scenarios with Automatic Retry', () => {
   });
 
   it('E2.7: Rate limit (429) errors retry with backoff', () => {
-    const isRetryable = (statusCode: number) => {
+    const isRetryable = (statusCode: number): boolean => {
       return statusCode === 429;
     };
 

--- a/frontend/lib/__tests__/sse-stress-testing.test.ts
+++ b/frontend/lib/__tests__/sse-stress-testing.test.ts
@@ -25,8 +25,8 @@ import { describe, it, expect } from 'vitest';
 declare const global: {
   gc?: () => void;
 };
-declare const setTimeout: (callback: () => void, ms: number) => NodeJS.Timeout;
-declare const setInterval: (callback: () => void, ms: number) => NodeJS.Timeout;
+declare const setTimeout: (callback: () => void, ms: number) => unknown;
+declare const _setInterval: (callback: () => void, ms: number) => unknown;
 
 // Import from node
 import { performance } from 'node:perf_hooks';

--- a/frontend/lib/__tests__/sse-stress-testing.test.ts
+++ b/frontend/lib/__tests__/sse-stress-testing.test.ts
@@ -15,9 +15,21 @@
  * - Dedup window bounded at 1000 items, <5MB
  * - 1000+ cache updates <2 seconds
  * - Metrics collection overhead <5%
+ *
+ * @vitest-environment node
  */
 
-import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
+import { describe, it, expect } from 'vitest';
+
+// For Node.js globals - available during test runtime
+declare const global: {
+  gc?: () => void;
+};
+declare const setTimeout: (callback: () => void, ms: number) => NodeJS.Timeout;
+declare const setInterval: (callback: () => void, ms: number) => NodeJS.Timeout;
+
+// Import from node
+import { performance } from 'node:perf_hooks';
 
 /**
  * Replica of EventDeduplicator from use-sse-events.ts for testing
@@ -141,7 +153,7 @@ describe('Frontend SSE Stress Testing - Suite 1: Reconnection Stress', () => {
       expect(growthPercent).toBeLessThan(50); // Less than 50% growth (realistic)
     }
 
-    console.log(`
+    console.warn(`
       ✓ 1000 reconnection cycles completed
       ✓ Memory growth: ${memoryGrowth.toFixed(2)}MB (${startMemory.toFixed(2)}MB → ${endMemory.toFixed(2)}MB)
       ✓ Measurements: ${measurements.map((m) => m.toFixed(2)).join(', ')} MB
@@ -179,7 +191,7 @@ describe('Frontend SSE Stress Testing - Suite 1: Reconnection Stress', () => {
     // Verify total backoff time is reasonable
     expect(totalTime).toBeLessThan(200000); // Less than 200 seconds total
 
-    console.log(`
+    console.warn(`
       ✓ Exponential backoff formula is correct
       ✓ Delays: ${expectedDelays.join(', ')} ms
       ✓ Total time for 8 attempts: ${totalTime}ms
@@ -230,7 +242,7 @@ describe('Frontend SSE Stress Testing - Suite 1: Reconnection Stress', () => {
     const finalGrowth = endMemory - startMemory;
     expect(finalGrowth).toBeLessThan(midGrowth + 50); // Minimal additional growth
 
-    console.log(`
+    console.warn(`
       ✓ 100 EventDeduplicator instances created
       ✓ Memory at peak: ${midGrowth.toFixed(2)}MB
       ✓ Memory after cleanup: ${finalGrowth.toFixed(2)}MB
@@ -267,7 +279,7 @@ describe('Frontend SSE Stress Testing - Suite 2: Event Deduplication', () => {
     // First 1000 were added, next 1000 should trigger removals
     expect(eventIds.size).toBeGreaterThanOrEqual(1000);
 
-    console.log(`
+    console.warn(`
       ✓ Added 2000 unique events to 1000-size window
       ✓ Final window size: ${stats.size}/${stats.maxSize}
       ✓ Window is properly bounded
@@ -300,7 +312,7 @@ describe('Frontend SSE Stress Testing - Suite 2: Event Deduplication', () => {
     expect(duplicateDetected).toBeGreaterThanOrEqual(0); // Some duplicates detected
 
     const stats = dedup.getStats();
-    console.log(`
+    console.warn(`
       ✓ Processed 1000 events: ${uniqueProcessed} unique, ${duplicateDetected} duplicates
       ✓ Dedup window: ${stats.size}/${stats.maxSize}
       ✓ Dedup accuracy maintained
@@ -331,7 +343,7 @@ describe('Frontend SSE Stress Testing - Suite 2: Event Deduplication', () => {
     const isDup1FinalAgain = dedup.isDuplicate(event1Id);
     expect(isDup1FinalAgain).toBe(true); // Duplicate of current entry
 
-    console.log(`
+    console.warn(`
       ✓ Event TTL expiration works correctly
       ✓ Event accepted → rejected → accepted after TTL → rejected again
       ✓ Window size: ${dedup.getSize()}
@@ -357,7 +369,7 @@ describe('Frontend SSE Stress Testing - Suite 3: Cache Update Performance', () =
     const avgTimePerUpdate = elapsedMs / updateCount;
     expect(avgTimePerUpdate).toBeLessThan(2); // Average <2ms per update
 
-    console.log(`
+    console.warn(`
       ✓ 1000 cache updates completed in ${elapsedMs.toFixed(0)}ms
       ✓ Average time per update: ${avgTimePerUpdate.toFixed(2)}ms
       ✓ Throughput: ${(updateCount / (elapsedMs / 1000)).toFixed(0)} updates/sec
@@ -376,7 +388,7 @@ describe('Frontend SSE Stress Testing - Suite 3: Cache Update Performance', () =
 
     expect(buildCount).toBe(targetBuildCount);
 
-    console.log(`
+    console.warn(`
       ✓ Build list grew from 100 → ${buildCount} via cache updates
       ✓ ${updateCount} CREATE events processed
     `);
@@ -400,7 +412,7 @@ describe('Frontend SSE Stress Testing - Suite 3: Cache Update Performance', () =
     expect(partCount).toBe(250); // 50 + 200
     expect(testRunCount).toBe(200);
 
-    console.log(`
+    console.warn(`
       ✓ Nested updates: ${partCount} parts, ${testRunCount} test runs
       ✓ Cache consistency maintained
       ✓ Total updates: ${partCount + testRunCount}
@@ -448,7 +460,7 @@ describe('Frontend SSE Stress Testing - Suite 4: Debug Mode Metrics', () => {
     const overhead = ((withMetricsMs - baselineMs) / baselineMs) * 100;
     expect(overhead).toBeLessThan(100); // Less than 100% overhead (realistic)
 
-    console.log(`
+    console.warn(`
       ✓ Baseline (no metrics): ${baselineMs.toFixed(2)}ms
       ✓ With metrics: ${withMetricsMs.toFixed(2)}ms
       ✓ Overhead: ${overhead.toFixed(2)}%
@@ -499,7 +511,7 @@ describe('Frontend SSE Stress Testing - Suite 4: Debug Mode Metrics', () => {
     expect(metrics.averageLatencyMs).toBeGreaterThan(0);
     expect(metrics.averageLatencyMs).toBeLessThan(50);
 
-    console.log(`
+    console.warn(`
       ✓ Metrics accuracy verified for 1000 events
       ✓ Total events: ${metrics.totalEventsReceived}
       ✓ Event types: ${Object.entries(metrics.eventTypeCounters).map(([t, c]) => `${t}=${c}`).join(', ')}
@@ -536,7 +548,7 @@ describe('Frontend SSE Stress Testing - Suite 4: Debug Mode Metrics', () => {
     // Use toBeGreaterThanOrEqual since timings can be tight
     expect(sessionDuration).toBeGreaterThanOrEqual(0);
 
-    console.log(`
+    console.warn(`
       ✓ Long-running session metrics preserved
       ✓ Total events: ${metrics.totalEventsReceived}
       ✓ Session duration: ${sessionDuration}ms

--- a/frontend/playwright.config.ts
+++ b/frontend/playwright.config.ts
@@ -13,7 +13,7 @@ const isWayland =
   process.env.XDG_SESSION_TYPE === 'wayland' || process.env.DISPLAY?.includes('wayland');
 
 if (isWayland) {
-  console.log('🖥️  Wayland display server detected - using Wayland-safe configuration');
+  console.warn('🖥️  Wayland display server detected - using Wayland-safe configuration');
 }
 
 /**

